novawallet/Common/Substrate/BlockTime/ChainTimelineFacade.swift

```diff
@@ -41,4 +42,36 @@ extension ChainTimelineFacade: ChainTimelineFacadeProtocol {
             return .createWithError(error)
         }
     }
+
+    func createTimepointThreshold(backIn days: Int) -> CompoundOperationWrapper<TimepointThreshold?> {
```

Comment:
I wouldn't complicate more timeline facade and revert BlockTimeEstimationService. I think the problem is that we need a way to subscribe to a threshold and fetch data based on it once on change. I think we could create a small service inheriting from `BaseObservableStateStore`. It decides based on chain whether to subscribe to a block number or create threshold immediately based on timestamp, then save it to the state. Interactors can start the service in setup and subscribe to it's state.